sdk/contentsafety/ai-content-safety-rest/test/public/contentSafety.spec.ts

```diff
@@ -0,0 +1,234 @@
+  });
+
+  it("analyze text", async function () {
+    const response = await client.path("/text:analyze").post({
+      body: {
+        text: "You are an idiot",
+        categories: ["Hate"],
+      },
+    });
+    if (isUnexpected(response)) {
+      throw new Error(response.body?.error.message);
+    }
+    assert.strictEqual(response.status, "200");
+    assert.equal(response.body.hateResult?.category, "Hate");
+    assert.notExists(response.body.selfHarmResult);
+  });
+
+  it("analyze image", async function () {
+    let base64Image: string;
+    if (isBrowser) {
+      const imagePath = "/base/samples-dev/example-data/image.png";
+      const response = await fetch(imagePath);
+      const buffer = await response.arrayBuffer();
+      const binary = new Uint8Array(buffer);
+      base64Image = uint8ArrayToBase64(binary);
+    } else {
+      const imagePath = path.join("samples-dev", "example-data", "image.png");
+      const buffer = fs.readFileSync(imagePath);
+      base64Image = buffer.toString("base64");
+    }
+    const response = await client.path("/image:analyze").post({
+      body: {
+        image: {
+          content: base64Image,
+        },
+        categories: ["Sexual"],
+      },
+    });
+    if (isUnexpected(response)) {
+      throw new Error(response.body?.error.message);
+    }
+    assert.strictEqual(response.status, "200");
+    assert.equal(response.body.sexualResult?.category, "Sexual");
+    assert.notExists(response.body.violenceResult);
+  });
+
+  it("create blocklist", async function () {
+    const createBlockListResponse = await client
+      .path("/text/blocklists/{blocklistName}", blocklistName)
+      .patch({
+        contentType: "application/merge-patch+json",
+        body: {
+          description: "test",
+        },
+      });
+    if (isUnexpected(createBlockListResponse)) {
+      throw new Error(createBlockListResponse.body?.error.message);
+    }
+    assert.strictEqual(createBlockListResponse.status, "201");
+    assert.equal(createBlockListResponse.body.blocklistName, blocklistName);
+  });
+
+  it("add block items", async function () {
+    const addBlockItemsResponse = await client
+      .path("/text/blocklists/{blocklistName}:addBlockItems", blocklistName)
+      .post({
+        body: {
+          blockItems: [
+            {
+              description: "Test block item 1",
+              text: blockItemText1,
+            },
+            {
+              description: "Test block item 2",
+              text: blockItemText2,
+            },
+            {
+              description: "Test block item 3",
+              text: blockItemText3,
+            },
+          ],
+        },
+      });
+    if (isUnexpected(addBlockItemsResponse)) {
+      throw new Error(addBlockItemsResponse.body?.error.message);
+    }
+    assert.strictEqual(addBlockItemsResponse.status, "200");
+    assert.isArray(addBlockItemsResponse.body.value);
+
+    if (isRecordMode()) {
```

Comment:
I think you will probably need to use !isPlaybackMode here as there's live mode, we should probably still need to wait in live mode ?

Reply:
Yes, we need to wait in live mode.